PGM/src/main/java/tc/oc/pgm/mutation/types/kit/TeamChestMutation.java

```diff
@@ -0,0 +1,135 @@
+    final static int CHEST_SIZE = 27;
+
+    final Map<Party, Inventory> teamChests = new WeakHashMap<>();
+
+    final Optional<WoolMatchModule> oWmm;
+
+    public TeamChestMutation(Match match) {
+        super(match, false);
+        oWmm = Optional.ofNullable(match().getMatchModule(WoolMatchModule.class));
+        for (Party party : match().getParties()) {
+            if (party.isParticipatingType()) {
+                // Could the chest title be localized properly?
+                teamChests.put(party, match().getServer().createInventory(null, CHEST_SIZE));
+            }
+        }
+    }
+
+    @Override
+    public void kits(MatchPlayer player, List<Kit> kits) {
+        super.kits(player, kits);
+        kits.add(getKitForPlayer(player));
+    }
+
+    // Open shared inventory instead of placing the chest
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void onChestUse(PlayerInteractEvent event) {
+        if (event.getItem() == null) return;
+
+        if (!(event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK)) return;
+        if (event.getItem().getType() != TOOL_TYPE) return;
+
+        Player bukkitPlayer = event.getPlayer();
+        if (bukkitPlayer == null) return;
+        Optional<Inventory> oTeamInventory = getTeamsInventory(bukkitPlayer);
+
+        oTeamInventory.ifPresent(teamInventory -> {
+            event.setCancelled(true);
+            // If the item is in the off-hand slot, it wont get put back visually for the player without this.
+            if(event.getHand() == EquipmentSlot.OFF_HAND) event.getActor().updateInventory();
+            bukkitPlayer.openInventory(teamInventory);
+
+        });
+    }
+
+    @EventHandler(priority = EventPriority.LOWEST)
+    public void onInventoryClick(InventoryClickEvent event) {
+        if (event.getCurrentItem() == null) return;
+
+        // No putting evil items (ender chest, possibly wool) into the chest
+        Optional<Inventory> teamChest = getTeamsInventory(event.getActor());
+        if (teamChest.filter(teamInventory -> {
+            return teamInventory.equals(event.getView().getTopInventory());
+        }).isPresent() &&
+                isItemEvil(event.getCurrentItem())) {
+            event.setCancelled(true);
+            return;
+        }
+
+        // If normal right click, in their inventory, on the chest, then open shared inventory.
+        getTeamsInventory(event.getActor()).ifPresent(teamInventory -> {
+            if (event.getInventory().getType() == InventoryType.CRAFTING &&
+                    event.getCurrentItem().getType() == TOOL_TYPE &&
+                    event.getAction() == InventoryAction.PICKUP_HALF) {
+                event.setCancelled(true);
+                // This resets their mouse position annoyingly, but without it items can get stuck in places.
+                event.getActor().closeInventory();
```

Comment:
Since you use `event.getActor()` several times, you want to store that variable.